pyramid_tm/__init__.py

```diff
@@ -126,4 +136,18 @@ def includeme(config):
     - If none of the above conditions are True, the transaction will be
       committed (via ``transaction.commit()``).
     """
+    config.add_request_method(
+        'pyramid_tm.create_tm', name='transaction', reify=True,
+    )
     config.add_tween('pyramid_tm.tm_tween_factory', under=EXCVIEW)
+
+    def ensure():
+        manager_hook = config.registry.settings.get("tm.manager_hook")
+        if manager_hook is not None:
+            try:
+                manager_hook = resolver.maybe_resolve(manager_hook)
+                config.registry.settings["tm.manager_hook"] = manager_hook
+            except (ImportError, ValueError):
+                raise ConfigurationError("Unable to resolve tm.manager_hook")
+
+    config.action(None, ensure, order=10)
```

Comment:
There's no public API for doing what you just described. I guess you could change the settings after doing the include but that'd surprise me as well as most people I think. It's definitely inconsistent with behavior in other pyramid addons that expect settings to be set before you include. Since you have control of `settings` before creating a configurator I'd just argue that you fix it there until a real public api was added.